java-frontend/src/main/java/org/sonar/java/model/GeneratedFile.java

```diff
@@ -48,17 +47,17 @@
 
 public class GeneratedFile implements InputFile {
 
-  private static final Logger LOG = Loggers.get(GeneratedFile.class);
-
   private final Path path;
 
   @VisibleForTesting
   final List<SmapFile> smapFiles = new ArrayList<>();
+  private final InputFile jspSourceFile;
 
   private SourceMap sourceMap;
 
-  public GeneratedFile(Path path) {
+  public GeneratedFile(Path path, @Nullable InputFile jspSourceFile) {
```

Comment:
This should not be annotated with `@Nullable`. This can not happen in production, only in tests.